playground/backend/internal/preparers/java_preparators.go

```diff
@@ -40,47 +38,85 @@ const (
 		Prepare: removePublicClassModifier,
-		Args:    []interface{}{filePath, classWithPublicModifierPattern, classWithoutPublicModifierPattern},
+		Args:    []interface{}{builder.filePath, classWithPublicModifierPattern, classWithoutPublicModifierPattern},
 	}
-	changePackagePreparator := Preparator{
+	builder.AddPreparer(removePublicClassPreparer)
+	return builder
+}
+
+//WithPackageChanger adds preparer to change package
+func (builder *JavaPreparersBuilder) WithPackageChanger() *JavaPreparersBuilder {
+	changePackagePreparer := Preparer{
 		Prepare: changePackage,
-		Args:    []interface{}{filePath, packagePattern, importStringPattern},
+		Args:    []interface{}{builder.filePath, packagePattern, importStringPattern},
 	}
-	removePackagePreparator := Preparator{
+	builder.AddPreparer(changePackagePreparer)
+	return builder
+}
+
+//WithPackageRemover adds preparer to remove package
+func (builder *JavaPreparersBuilder) WithPackageRemover() *JavaPreparersBuilder {
+	removePackagePreparer := Preparer{
 		Prepare: removePackage,
-		Args:    []interface{}{filePath, packagePattern, newLinePattern},
+		Args:    []interface{}{builder.filePath, packagePattern, newLinePattern},
 	}
-	unitTestFileNameChanger := Preparator{
+	builder.AddPreparer(removePackagePreparer)
+	return builder
+}
+
+//WithFileNameChanger adds preparer to remove package
+func (builder *JavaPreparersBuilder) WithFileNameChanger() *JavaPreparersBuilder {
+	unitTestFileNameChanger := Preparer{
 		Prepare: changeJavaTestFileName,
-		Args:    []interface{}{filePath},
+		Args:    []interface{}{builder.filePath},
+	}
+	builder.AddPreparer(unitTestFileNameChanger)
+	return builder
+}
+
+// GetJavaPreparers returns preparation methods that should be applied to Java code
+func GetJavaPreparers(builder *PreparersBuilder, isUnitTest bool, isKata bool) {
+	if !isUnitTest && !isKata {
+		builder.JavaPreparers().
+			WithPublicClassRemover().
+			WithPackageChanger()
+	}
+	if isUnitTest {
+		builder.JavaPreparers().
+			WithPackageChanger().
+			WithFileNameChanger()
+	}
+	if isKata {
+		builder.JavaPreparers().
+			WithPublicClassRemover().
+			WithPackageRemover()
 	}
-	return &[]Preparator{removePublicClassPreparator, changePackagePreparator, removePackagePreparator, unitTestFileNameChanger}
 }
 
 //changePackage changes the 'package' to 'import' and the last directory in the package value to '*'
 func changePackage(args ...interface{}) error {
-	validationResults := args[3].(*sync.Map)
-	isKata, ok := validationResults.Load(validators.KatasValidatorName)
-	if ok && isKata.(bool) {
-		return nil
-	}
 	err := replace(args...)
 	return err
 }
 
 //removePackage remove the package line in the katas.
 func removePackage(args ...interface{}) error {
-	validationResults := args[3].(*sync.Map)
-	isKata, ok := validationResults.Load(validators.KatasValidatorName)
-	if ok && isKata.(bool) {
-		err := replace(args...)
-		return err
-	}
-	return nil
+	err := replace(args...)
+	return err
 }
```

Comment:
Do we even need this function if we can directly use `replace`?

Reply:
Removed